homeassistant/components/anthemav/config_flow.py

```diff
@@ -0,0 +1,109 @@
+import asyncio
+import logging
+from typing import Any
+
+import anthemav
+from anthemav.connection import Connection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import (
+    CONF_MODEL,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEVICE_TIMEOUT_SECONDS,
+    DOMAIN,
+    EMPTY_MAC,
+    UNKNOWN_MODEL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+
+async def connect_device(user_input: dict[str, Any]) -> Connection:
+    """Connect to the AVR device."""
+    avr = await anthemav.Connection.create(
+        host=user_input[CONF_HOST], port=user_input[CONF_PORT], auto_reconnect=False
+    )
+    await avr.reconnect()
+    await avr.protocol.wait_for_device_initialised(DEVICE_TIMEOUT_SECONDS)
+    return avr
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Anthem A/V Receivers."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        if CONF_NAME not in user_input:
+            user_input[CONF_NAME] = DEFAULT_NAME
+
+        errors = {}
+
+        avr: Connection | None = None
+        try:
+            avr = await connect_device(user_input)
+        except OSError:
+            _LOGGER.error(
+                "Can't established connection to %s:%s",
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+            )
+            errors["base"] = "cannot_connect"
+        except asyncio.TimeoutError:
+            errors["base"] = "cannot_receive_deviceinfo"
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.exception(err)
+            errors["base"] = "unknown"
+        else:
+            user_input[CONF_MAC] = format_mac(avr.protocol.macaddress)
+            user_input[CONF_MODEL] = avr.protocol.model
+            if (
+                user_input[CONF_MAC] == EMPTY_MAC
+                or user_input[CONF_MODEL] == UNKNOWN_MODEL
+            ):
+                _LOGGER.error("Invalid MacAddress or model")
+                errors["base"] = "cannot_receive_deviceinfo"
+            else:
+                await self.async_set_unique_id(user_input[CONF_MAC])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
```

Comment:
I guess the IP can change and thereby maybe save to options instead of data and implement OptionsFlow so the user can modify the IP if necessary?

Reply:
Sound good. I didn't know about OptionsFlow. Is it what would bring the "Configure" button in the UI of the integration? 
I'll look into this, you are right, it would be great to modify the IP.

Reply:
After looking closely to other integrations, it doesn't look like changing the IP Address in option flow is done by any. The problem is changing the IP address means you completely reconnect to the device and potentially need to change the uniqueid use for the integration (if you connect to a separate device it would be a different mac address). Because IP address is the only information needed for this integration, I don't think it is needed to add option flow. I'm planning to expand this library when this PR is out of the way and I would implement options flow for the additional options.

If you think I must implement OptionsFlow, do you know if it is possible to change the integration uniqueId in this case?

Reply:
I think you use the mac-address as the unique id so in a validation in optionsflow you might check that the mac-address comes back same as previously and then it's fine to change the IP address.
You don't have to have an OptionsFlow. Just a suggestion as otherwise if the IP change then the user needs to remove the integration and configure it from scratch (which is maybe not so big of a deal but don't know this integration).